File: Day01/main.py
with open('input_data.txt','r') as f:
    lines = f.readlines()
f.close()

elf_calories = [0]
elf_index = 0
for line in lines:
    if line == '\n': # spazio vuoto
        elf_index += 1
        elf_calories.append(0)
    else:
        elf_calories[elf_index] += int(line)

# part 1
elf_calories.sort()
elf_fat = elf_calories[-1]
print(elf_fat)
# part 2
elf_fat3 = elf_calories[-3] + elf_calories[-2] + elf_calories[-1]
print(elf_fat3)
    

File: Day01/main.m
clc; clear; close all;
%% take data
lines = readlines('input_data.txt');

%% convert data
most_calories = [0 0 0];
current_calories = 0;
for line = 1:length(lines)
    if lines(line) ~= ""
        current_calories = current_calories + str2double(lines(line));
    else
        if any(current_calories > most_calories)
            most_calories(4) = current_calories;
            most_calories = sort(most_calories,'descend');
            most_calories = most_calories(1:end-1);
        end
        current_calories = 0;
    end
end
fprintf(['Point 1: ' int2str(most_calories(1)) '\n'])
fprintf(['Point 2: ' int2str(sum(most_calories)) '\n'])